src/main/java/in/ashwanthkumar/gocd/slack/ruleset/PipelineRule.java

```diff
@@ -101,11 +112,12 @@ public PipelineRule setWebhookUrl(String webhookUrl) {
         return this;
     }
 
-    public boolean matches(String pipeline, String stage, String group, final String pipelineState) {
+    public boolean matches(String pipeline, String stage, String group, String label, final String pipelineState) {
         return pipeline.matches(nameRegex)
                 && stage.matches(stageRegex)
                 && matchesGroup(group)
-                && Iterables.exists(status, hasStateMatching(pipelineState));
+                && Iterables.exists(status, hasStateMatching(pipelineState))
+                || label.matches(labelRegex);
```

Comment:
if the labelRegex is optional, this should ideally be something like:

```suggestion
                && (StringUtils.isNotBlank(labelRegex) ? label.matches(labelRegex) || true);
```

Reply:
Changes done

Reply:
do you know what is the default regex if the labelRegex is not set in the config for a pipeline?

It will be a blank value or .*

I modified one the test case with blank value here: RuleTest.java Line number 36

Reply:
Thanks for the update. Turns out the default pipelineRule that is merged with is defined in this file: https://github.com/ashwanthkumar/gocd-slack-build-notifier/blob/master/src/main/resources/reference.conf#L32. Please consider adding the labelRegex with `.*` as the value so by default it would just match everything. And with that we can change this snippet back to just:

```suggestion
                && label.matches(labelRegex)
```

Sorry for the back and forth, but this should hopefully be the last comment and after that we should be able to merge this in.

Reply:
No need to apologize, @ashwanthkumar. I genuinely value the time you're dedicating to assist us, and I've made the requested changes based on your most recent comment.

Reply:
@ashwanthkumar Kindly re-review

Reply:
@ashwanthkumar Kindly re-review